### PR TITLE
Trained model & API update & tests update

### LINKED ISSUE #9: Implementation & Testing & Documentation
1. a complete implementation that includes the API endpoints and returns a prediction using our trained model.
2. an updated test suite for our microservice endpoints with newly added features
3. an updated swagger document with newly added features
4.  an updated README that clearly and concisely describes features used to training our model, and how our retrained model performs better than the baseline model.

### PR DESCRIPTION
**Pull request by lmxlucy**
1. Build a random forest classifier to predict whether an applicant is a high-quality student using training data with these variables: 'studytime', 'failures', 'absences', 'activities', 'internet', 'Medu', 'Fedu', 'G1', 'G2'
2. Tested the accuracy of our model using testing data: the accuracy improved from 53.57% (baseline) to 96.00%.
3. Updated API endpoints to take in the variables mentioned above
4. Added error handling to API
5. Updated pytest according to variable changes

### CODE REVIEW

**Comment by YonahGoldberg** [comment]
Why does it return 1 or 0, is there whether or not the G3 prediction is 15 or over? I thought we were going to return the actual G3 prediction.

I also tried to test it in the Jupyter notebook, but I'm getting ModuleNotFound errors, not sure if I'm doing something wrong.

Other than that, I think we should add some more descriptive error messages instead of just the bad requests. Although that can go in a later PR. Oh, and we should update the Swagger docs.

**Comment by lmxlucy** [comment]
> Why does it return 1 or 0, is there whether or not the G3 prediction is 15 or over? I thought we were going to return the actual G3 prediction.
I think the goal is to predict whether the candidate is a high quality student. I believe a boolean is more useful than giving a score. 

> I also tried to test it in the Jupyter notebook, but I'm getting ModuleNotFound errors, not sure if I'm doing something wrong.
I have no problem testing it on mine. I did it in Codespace. The dependency setup is like a hell and I couldn't get it right on my local machine. Codespace works very well though. You can probably try codespace and we will ask others to test it too.

> Other than that, I think we should add some more descriptive error messages instead of just the bad requests. Although that can go in a later PR. Oh, and we should update the Swagger docs.
I can add more descriptive error message later. I updated the swagger docs in another PR. Please go review!! Thanks!

**Review by YonahGoldberg — APPROVED** [review]
Looks good!

**Review by mzhou08 — APPROVED** [review]
lgtm!

File: app/handlers/routes.py [hunk 1]
```diff
@@ -19,16 +19,21 @@ def hello():
...
-
+        activities = request.args.get('activities')
+        internet = request.args.get('internet')
+        medu = request.args.get('Medu')
+        fedu = request.args.get('Fedu')
+        g1 = request.args.get('G1')
+        g2 = request.args.get('G2')
+        if not (studytime and failures and absences and activities and internet and medu and fedu and g1 and g2):
+            abort(400)
+        if not (studytime.isnumeric() and failures.isnumeric() and absences.isnumeric() and activities.isnumeric() and internet.isnumeric() and medu.isnumeric() and fedu.isnumeric() and g1.isnumeric() and g2.isnumeric()):
```
Comment (mzhou08) [inline app/handlers/routes.py:33]: Can we add range checks for the int values?

**Pull request merged**